### ISSUE TITLE
Having an issue to use the framework to train on a new domain

### ISSUE THREAD

**Issue by bimhud** [WING-NUS/JD2Skills-BERT-XMLC#6]
Hi @knmnyn,
Not sure what I was doing wrong. 
I'm currently using a new domain of text data, it is unable to start the training due to the checkpoint model is attached with 2548 labels.  When changing these number, the pre-train model file is not accepted. Just wondering if it could be modified to be used with a different number of labels?

Thanks.

**Comment by knmnyn** [comment]
Hi @bimhud : I don't believe it can be used with a different number of labels.  You may need to retrain the model if you need to use new labels.  @AKBakshay may be able to elaborate; @AKBakshay can you answer and close the thread when you've replied?

**Comment by bimhud** [comment]
Hi @knmnyn @AKBakshay , I did re-train the model by replaced the downloaded checkpoint model with the original one such as https://s3.amazonaws.com/models.huggingface.co/bert/bert-base-uncased-pytorch_model.bin, voca and config files

From my understanding, this will help fine-tune the two last layers using the same framework. It is working without any issue. However, the performance is not as good as your checkpoint and the prediction result doesn't make sense. 

Just wondering how many epoch that was trained in order to produce your 'checkpoint' models and what are the original pre-train model that you were using.

Any suggestion would be really appreciated.

**Comment by AKBakshay** [comment]
@knmnyn Thank you for the input!  

@bimhud In the repo, the available model weights correspond to the model with 2548 labels as output. Based on the requirement, the output size can be changed. But doing that, the provided weights can't be loaded. 

You can manually make changes to load the weights of the provided model, remove its last 2 layers and add new layers, based on the requirement.
